django_filters/filters.py

```diff
@@ -235,6 +240,12 @@ def filter(self, qs, value):
 
         return qs.distinct() if self.distinct else qs
 
+    def get_filter_predicate(self, v):
+        try:
+            return {self.name: getattr(v, self.field.to_field_name)}
+        except AttributeError:
+            return {self.name: v}
```

Comment:
What I mean is that the common code path is the fallback (`except`), and therefor a `hasattr` check (instead of the `try`, not instead of the `getattr`) might be better in the end.
I agree though that we should keep it like this for now.